fixtures/workers-with-assets/tests/index.test.ts

```diff
@@ -73,44 +77,103 @@ describe("[Workers + Assets] `wrangler dev`", () => {
+		let response = await fetch(`http://${ip}:${port}/index.html`, {
 			method: "HEAD",
 		});
 		expect(response.status).toBe(405);
 		expect(response.statusText).toBe("Method Not Allowed");
 
-		response = await fetch(`http://${ip}:${port}/hello.html`, {
+		response = await fetch(`http://${ip}:${port}/index.html`, {
 			method: "POST",
 		});
 		expect(response.status).toBe(405);
 		expect(response.statusText).toBe("Method Not Allowed");
 
-		response = await fetch(`http://${ip}:${port}/hello.html`, {
+		response = await fetch(`http://${ip}:${port}/index.html`, {
 			method: "PUT",
 		});
 		expect(response.status).toBe(405);
 		expect(response.statusText).toBe("Method Not Allowed");
 
-		response = await fetch(`http://${ip}:${port}/hello.html`, {
+		response = await fetch(`http://${ip}:${port}/index.html`, {
 			method: "DELETE",
 		});
 		expect(response.status).toBe(405);
 		expect(response.statusText).toBe("Method Not Allowed");
 
-		response = await fetch(`http://${ip}:${port}/hello.html`, {
+		response = await fetch(`http://${ip}:${port}/index.html`, {
 			method: "OPTIONS",
 		});
 		expect(response.status).toBe(405);
 		expect(response.statusText).toBe("Method Not Allowed");
 
-		response = await fetch(`http://${ip}:${port}/hello.html`, {
+		response = await fetch(`http://${ip}:${port}/index.html`, {
 			method: "PATCH",
 		});
 		expect(response.status).toBe(405);
 		expect(response.statusText).toBe("Method Not Allowed");
 	});
+
+	it("should forward all request types to the user worker if there are *not* assets on that route", async ({
+		expect,
+	}) => {
+		// Unlike above, if the AW does NOT find assets on a route, non-GET request should return 404s
+		// This is because all requests are first sent to the AW and only then to the UW
+		// and we don't want to 405 on a valid POST request intended for the UW
+		// POST etc. request -> RW -> AW -> checks manifest --404--> RW -> UW -> response
+
+		let response = await fetch(`http://${ip}:${port}/no-assets-here`, {
+			method: "GET",
+		});
+		expect(response.status).toBe(200);
+
+		response = await fetch(`http://${ip}:${port}/no-assets-here`, {
+			method: "HEAD",
+		});
+		expect(response.status).toBe(200);
+
+		response = await fetch(`http://${ip}:${port}/no-assets-here`, {
+			method: "POST",
+		});
+		expect(response.status).toBe(200);
+
+		response = await fetch(`http://${ip}:${port}/no-assets-here`, {
+			method: "PUT",
+		});
+		expect(response.status).toBe(200);
+
+		response = await fetch(`http://${ip}:${port}/no-assets-here`, {
+			method: "DELETE",
+		});
+		expect(response.status).toBe(200);
+
+		response = await fetch(`http://${ip}:${port}/no-assets-here`, {
+			method: "OPTIONS",
+		});
+		expect(response.status).toBe(200);
+
+		response = await fetch(`http://${ip}:${port}/no-assets-here`, {
+			method: "PATCH",
+		});
+		expect(response.status).toBe(200);
+	});
+
+	it("should be able to use an ASSETS binding", async ({ expect }) => {
```

Comment:
we should also test the use case when an asset is actually found in the asset manifest

Reply:
turns out that's already tested. moving on